Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/EnumsAndHelperStructs.proto

```diff
@@ -0,0 +1,1061 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+syntax = "proto3";
+
+import "tools/google/protobuf/timestamp.proto";
+import "tools/google/protobuf/duration.proto";
+
+package BuildXL.Execution.Analyzer;
+
+option csharp_namespace = "BuildXL.Execution.Analyzer.Xldb";
+
+//
+// Enums
+//
+
+// Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, 
+// not children of it. Therefore, enum values must be unique within "BuildXL.Execution.Analyzer", not 
+// just within a particular enum. If there are conflicting enum values, append the name of the enum in front.
+// ie: ChooseWorkerCpu ---> PipExecutionStep_ChooseWorkerCpu
+
+enum ExecutionEventId{
+    // Since we are using ExecutionEventId as in the key and protobuf 
+    // does not serialize default values, the pattern matching was bugging out. 
+    // A solution is to have an "invalid" enum constant take position 0. 

```

Comment:
I think I commented re this in past but I want to bring it up again:
I think we might want to add that 'invlid = 0' to all objects. If one day we decide that we want to use something as a key, we'll need to move all the fields, and I'll be a breaking change. Alternatively, maybe create objects that are exclusively used as keys?